### PR TITLE
create dummy assigned story_translations_table (Taylor's version)

### PR DESCRIPTION
**Pull request by andrewyguo**
build back better

## Notion ticket link

[Dummy Assigned Stories Table](https://www.notion.so/uwblueprintexecs/Create-dummy-Assigned-Story-Translations-table-7967d4c1605749498d1d4529dc0c3de0)

## Implementation description

- Create new table component to display Assigned Story Translations for a user on user page 
- Create new GQL query string to get story translations by user id 

## Steps to test

1. Go to admin page and click on any user 
2. Ensure that their story translations / reviews are properly displayed 

## What should reviewers focus on?

- Styling is correct, entries in table are correct 

## Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] For backend changes, I have run the appropriate linters: `docker exec -it planet-read_py-backend_1 /bin/bash -c "black . && isort --profile black ."` and I have generated new migrations: `flask db migrate -m "<your message>"`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/admin/AssignedStoryTranslationsTable.tsx [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+            onClick={() => sort("language")}
+          >{`LANGUAGE & LEVEL ${isAscendingLanguage ? "↑" : "↓"}`}</Th>
+          <Th cursor="pointer" onClick={() => sort("stage")}>{`PROGRESS ${
+            isAscendingStage ? "↑" : "↓"
+          }`}</Th>
+          <Th>ACTION</Th>
+        </Tr>
+      </Thead>
+      <Tbody>{tableBody}</Tbody>
+      {/* <hr marginTop="-10px"/>  How do I add a horizontal line lol */}
```
Comment (andrewyguo) [inline frontend/src/components/admin/AssignedStoryTranslationsTable.tsx:162]: the figma has a light grey horizontal line here, not too sure how to properly implement this

File: frontend/src/components/admin/AssignedStoryTranslationsTable.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    isAscendingStage
+      ? t1.stage.localeCompare(t2.stage)
+      : t2.stage.localeCompare(t1.stage);
+
+  const sort = (field: String) => {
+    const newStoryTranslations = [...storyTranslations];
+    switch (field) {
+      case "title": {
+        setIsAscendingTitle(!isAscendingTitle);
+
```
Comment (gaoxk) [inline frontend/src/components/admin/AssignedStoryTranslationsTable.tsx:71]: nit: this empty line bothers me for some reason LOL did this come from the linter?

File: frontend/src/components/admin/AssignedStoryTranslationsTable.tsx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+      : t2.language.localeCompare(t1.language);
+
+  const stageSort = (t1: StoryTranslation, t2: StoryTranslation) =>
+    isAscendingStage
+      ? t1.stage.localeCompare(t2.stage)
+      : t2.stage.localeCompare(t1.stage);
+
+  const sort = (field: String) => {
+    const newStoryTranslations = [...storyTranslations];
+    switch (field) {
```
Comment (gaoxk) [inline frontend/src/components/admin/AssignedStoryTranslationsTable.tsx:68]: nit: I think you could simplify this with a dictionary mapping `field : String -> [sortFn, setIsAscendingState, isAscendingState]`. Then this fn could be simplified to 
```
const newStoryTranslations = [...storyTranslations];
const setIsAscendingState, sortFn = fieldSortDict[field];
setIsAscendingState(!isAscendingState);
newStoryTranslations.sort(sortFn);
setStoryTranslations(newStoryTranslations);
```
up to you if you wanna do this change

File: frontend/src/components/admin/AssignedStoryTranslationsTable.tsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+  const tableBody = storyTranslations.map(
+    (translation: StoryTranslation, index: number) => (
+      <Tr
+        borderBottom={
+          index === storyTranslations.length - 1 ? "1em solid transparent" : ""
+        }
+        key={`${translation?.storyTranslationId}`}
+      >
+        <Td>{translation.title}</Td>
```
Comment (gaoxk) [inline frontend/src/components/admin/AssignedStoryTranslationsTable.tsx:108]: nit can you link the title like so https://github.com/uwblueprint/planet-read/blob/main/frontend/src/components/admin/StoryTranslationsTable.tsx#L79-L84

File: frontend/src/components/admin/AssignedStoryTranslationsTable.tsx (hunk 1 above)
Comment (gaoxk) [reply frontend/src/components/admin/AssignedStoryTranslationsTable.tsx:162]: I wouldnt worry about the horizontal line - imo the alternating cell shading makes it quite clear that the button is unrelated to any given row

Comment (gaoxk) [reply frontend/src/components/admin/AssignedStoryTranslationsTable.tsx:162]: lets see what the designers say tho

File: frontend/src/components/admin/AssignedStoryTranslationsTable.tsx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+  storyTranslations,
+  setStoryTranslations,
+  userId,
+}: AssignedStoryTranslationsTableProps) => {
+  const [isAscendingTitle, setIsAscendingTitle] = useState(true);
+  const [isAscendingRole, setIsAscendingRole] = useState(true);
+  const [isAscendingLanguage, setIsAscendingLanguage] = useState(true);
+  const [isAscendingStage, setIsAscendingStage] = useState(true);
+
+  const getRole = (translation: StoryTranslation) => {
```
Comment (gaoxk) [inline frontend/src/components/admin/AssignedStoryTranslationsTable.tsx:36]: all this sorting stuff reaaaally lengthens the file length. Thoughts on making a new `utils/` file and moving all this sort stuff to it?

File: frontend/src/components/admin/AssignedStoryTranslationsTable.tsx [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+        </Tr>
+      </Thead>
+      <Tbody>{tableBody}</Tbody>
+      {/* <hr marginTop="-10px"/>  How do I add a horizontal line lol */}
+      <Button
+        variant="blueOutline"
+        size="secondary"
+        margin="5px 0px 15px 10px"
+        width="200px"
+        border="2px"
```
Comment (gaoxk) [inline frontend/src/components/admin/AssignedStoryTranslationsTable.tsx:168]: why set the border style?

File: frontend/src/components/admin/AssignedStoryTranslationsTable.tsx [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+          >{`LANGUAGE & LEVEL ${isAscendingLanguage ? "↑" : "↓"}`}</Th>
+          <Th cursor="pointer" onClick={() => sort("stage")}>{`PROGRESS ${
+            isAscendingStage ? "↑" : "↓"
+          }`}</Th>
+          <Th>ACTION</Th>
+        </Tr>
+      </Thead>
+      <Tbody>{tableBody}</Tbody>
+      {/* <hr marginTop="-10px"/>  How do I add a horizontal line lol */}
+      <Button
```
Comment (gaoxk) [inline frontend/src/components/admin/AssignedStoryTranslationsTable.tsx:163]: nit alphabetically sort these props

**Review by gaoxk — CHANGES_REQUESTED** [review]
noicccccccccccccccce 😎

File: frontend/src/components/admin/AssignedStoryTranslationsTable.tsx (hunk 5 above)
Comment (andrewyguo) [reply frontend/src/components/admin/AssignedStoryTranslationsTable.tsx:36]: Hmm the sorting functions currently use a state variable `isAscending...` and also use the function `getRole` on the `roleSort` function. Moving all of these to a utils folder would be a bit messy. I think it'll look better in here

File: frontend/src/components/admin/AssignedStoryTranslationsTable.tsx (hunk 6 above)
Comment (andrewyguo) [reply frontend/src/components/admin/AssignedStoryTranslationsTable.tsx:168]: The border was really thin so I added the additionally styling to get it to match with the figma better

File: frontend/src/components/admin/AssignedStoryTranslationsTable.tsx [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+    }
+    return "Error";
+  };
+
+  const roleSort = (t1: StoryTranslation, t2: StoryTranslation) =>
+    isAscendingRole
+      ? getRole(t1).localeCompare(getRole(t2))
+      : getRole(t2).localeCompare(getRole(t1));
+
+  const sortDict: AssignedStoryTranslationsFieldSortDict = {
```
Comment (gaoxk) [inline frontend/src/components/admin/AssignedStoryTranslationsTable.tsx:61]: 😎 🚀 👌 🔥 🔥 😃

**Review by gaoxk — APPROVED** [review]
nice 😎

File: frontend/src/components/admin/AssignedStoryTranslationsTable.tsx (hunk 8 above)
Comment (andrewyguo) [reply frontend/src/components/admin/AssignedStoryTranslationsTable.tsx:61]: whatever boomer

**Pull request merged**